k8s/tools/deploy-pubsub-gke.sh

```diff
@@ -0,0 +1,262 @@
+  echo -n "Do you want to use $DEVSHELL_PROJECT_ID as the target project? (y / n) > "
+  read response
+  if [[ $response != "y" && $response != "Y" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+fi
+
+# Check if the configured VPC network exists.
+networks=$(gcloud -q compute networks list --filter="name=$VPC_NETWORK" |wc -l)
+if [[ "${networks}" -lt "2" ]]; then
+        echo "ERROR: VPC network $VPC_NETWORK not found, please create this first."
+        exit 1
+fi
+
+# Enable IAM services
+gcloud -q services --project $DEVSHELL_PROJECT_ID enable iam.googleapis.com
+
+# Use local `gcloud auth` credentials rather than creating new Service Account.
+if [[ "$*" == *--no-gcloud-auth* ]] ; then
+  SA_NAME="turbinia"
+  SA_MEMBER="serviceAccount:$SA_NAME@$DEVSHELL_PROJECT_ID.iam.gserviceaccount.com"
+
+  if ! gcloud --project $DEVSHELL_PROJECT_ID iam service-accounts list |grep $SA_NAME; then
+    # Create service account
+    gcloud --project $DEVSHELL_PROJECT_ID iam service-accounts create "${SA_NAME}" --display-name "${SA_NAME}"
+  fi
+
+  # Grant IAM roles to the service account
+  echo "Grant permissions on service account"
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/cloudfunctions.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/cloudsql.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/compute.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/container.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/datastore.indexAdmin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/editor'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/logging.logWriter'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/pubsub.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/redis.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/servicemanagement.admin'
+  gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/storage.admin'
+
+  # Create and fetch the service account key
+  echo "Fetch and store service account key"
+  gcloud --project $DEVSHELL_PROJECT_ID iam service-accounts keys create ~/key.json --iam-account "$SA_NAME@$DEVSHELL_PROJECT_ID.iam.gserviceaccount.com"
+  export GOOGLE_APPLICATION_CREDENTIALS=~/key.json
+
+# TODO: Do real check to make sure credentials have adequate roles
+elif [[ $( gcloud auth list --filter="status:ACTIVE" --format="value(account)" | wc -l ) -eq 0 ]] ; then
+  echo "No gcloud credentials found.  Use 'gcloud auth login' and 'gcloud auth application-default' to log in"
```

Comment:
I know this comes from the deploy.sh script, but I think this is supposed to be `gcloud auth application-default login`.

Reply:
I added `gcloud auth login <EMAIL> or as either method should work

Reply:
I think we actually need both here AFAIK -- What I meant to point out is that I believe the `... application-default` command needs the `login` at the end of it.

Reply:
Ahhh okay got it, missed that part.

Just fixed by adding in `login` !